### PR TITLE
Weighted graphs

### PR DESCRIPTION
**Pull request by mlmenard**
* Support for Weighted graphs (new traits and some basic tests).
* Cargo clippy changes.
* Fractional Coreness Implementation

### CODE REVIEW

File: src/dachshund/algorithms/clustering.rs [hunk 1]
```diff
@@ -49,7 +49,6 @@ pub trait Clustering: GraphBase {
         let ordered_nodes = self
             .get_nodes_iter()
             .filter(|node| node.degree() >= 2)
-            .map(|node| node)
```
Comment (zellski) [inline src/dachshund/algorithms/clustering.rs:52]: Was this ever not a NOOP?

File: src/dachshund/node.rs [hunk 2]
```diff
@@ -180,10 +209,7 @@ impl NodeBase for SimpleNode {
     /// used to determine degree in a subgraph (i.e., the clique we're considering).
     /// HashSet is supplied by Candidate struct.
     fn count_ties_with_ids(&self, ids: &HashSet<NodeId>) -> usize {
-        ids.iter()
-            .filter(|x| self.neighbors.contains(x))
-            .collect::<Vec<&NodeId>>()
-            .len()
+        ids.iter().filter(|x| self.neighbors.contains(x)).count()
```
Comment (zellski) [inline src/dachshund/node.rs:212]: Ahh, what a pleasure.

File: src/dachshund/weighted_undirected_graph.rs [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        // [TODO Fix trait bounds and move this to coreness.rs.]
+        // [TODO:Perf] Switch to hashbrown. Benchmark performance.
+        let mut pq = PriorityQueue::with_capacity(self.nodes.len());
+
+        for node in self.get_nodes_iter() {
+            pq.push(node.get_id(), Reverse(NotNan::new(node.weight()).unwrap()));
+        }
+        let mut coreness: HashMap<NodeId, f64> = HashMap::new();
+
+        while !pq.is_empty() {
```
Comment (zellski) [inline src/dachshund/weighted_undirected_graph.rs:128]: I flipped into "expect documentation" mode thanks to the comment above that begins with "Start by making a priority queue", so around this point I start feeling bereft & looking for another helpful comment explaining what's about to happen with coreness shells :)

**Review by zellski — APPROVED** [review]
Fabulous. I didn't spot anything amiss herein, with the caveat that I haven't worked with either the cliquerator or Rust in a year, so it's possible you'd want a second set of eyes.

File: src/dachshund/weighted_undirected_graph.rs (hunk 3 above)
Comment (mlmenard) [reply src/dachshund/weighted_undirected_graph.rs:128]: Agreed that the documentation here totally sucks. I added some more comments and will add that to this PR before merging.

**Pull request merged**